src/StructureFactors.jl

```diff
@@ -931,3 +931,75 @@ function sf_slice(sf::StructureFactor, points::Vector;
     return slice
 end
 
+
+
+#========== Powder Averaging WIP ==========#
+""" gaussian_smoothing(k, q, S, L)
+    Returns 'Powder-Averaged' contribution to different bins according to 
+the exact formula for Gaussian Smoothing Kernel for each S(q). 
+
+Args 
+    k :: Range of 'bins' for Powder Averaging. 
+    q :: Magnitude of 'q' vector; Gaussian smoothing only a function of magnitude. 
+    S :: Intensity at 'q' vector. 
+    L :: Length-scale for smoothing kernel; suggested maximum(Lx, Ly, Lz). 
+        User can control this but issues may arise if σ for Gaussian Smoothing kernel is too big or small w.r.t spacing between q-vectors. 
+"""
+function gaussian_smoothing(k, q, S, L)
+    sig2 = (2π/L)^2.0 
+    c = (2π*sig2)^(3/2)
+    exp_pos = exp.( (.-k.*k.-q.*q.+2.0.*k.*q) ./ (2.0.*sig2) )
+    exp_neg = exp.( (.-k.*k.-q.*q.-2.0.*k.*q) ./ (2.0.*sig2) )
+    if q == 0 # Regularization 
+        return (exp_pos.-exp_neg).*(sig2.*S) ./ (2.0.*k.*(q+1e-7).*c)
+    else 
+        return (exp_pos.-exp_neg).*(sig2.*S) ./ (2.0.*k.*q.*c)
+    end
+end 
+
+
+
+""" powder_avg(ssf) 
+Computes Powder Average from Crystal structure factor. 
+Args
+    ssf :: Sunny "static" structure factor object, after diple and phase factor has been applied. 
+
+Returns 
+    - K :: bins for Powder Avearge. 
+    - IQ :: Intensity for each 'K' bins. 
+
+Users may have control over
+    - K :: number/resolutions of bins for momentum transfer magnitude. 
+    - Other Kernels beyond Gaussian. 
+    - σ for Gaussian Kernel. (can break Gaussian Kernel. )
+    - normalization for Intensity. (Easy to Apply afterwards as needed. )
+"""
+function powder_avg(ssf) 
+    
+    # Get q-vector values 
+	q_vals = Sunny.q_labels(ssf) 
```

Comment:
Misaligned indentation occurs because of a mix of tabs and spaces (in your editor, you may have set a tab character to display as four spaces). Are you using VSCode?